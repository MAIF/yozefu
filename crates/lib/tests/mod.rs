use insta::{assert_debug_snapshot, glob};
use rdkafka::message::OwnedMessage;
use serde::Deserialize;
use std::fs;
use tokio::runtime::Runtime;
use yozefu_lib::{ExportedKafkaRecord, KafkaRecord, parse_search_query};

#[test]
fn test_inputs() {
    unsafe {
        use std::env;
        // Set the timezone to Paris to have a fixed timezone for the tests
        env::set_var("TZ", "Europe/Paris");
    }
    glob!("inputs/search-queries/*.sql", |path| {
        let input = fs::read_to_string(path).unwrap();
        let input = input.trim();
        insta::with_settings!({
            description => input.replace("\n", " "),
            filters => vec![
            ("[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}\\.[0-9]{6,}\\+[0-9]{2}:[0-9]{2}", "[datetime]"),
        ]}, {
            assert_debug_snapshot!(parse_search_query(input));
        });
    });
}

#[test]
fn test_exported_record() {
    glob!("inputs/parsed-records/record*.json", |path| {
        let input = fs::read_to_string(path).unwrap();
        let record: KafkaRecord = serde_json::from_str(&input).unwrap();
        assert_debug_snapshot!(ExportedKafkaRecord::from(&record));
    });
}

#[test]
fn test_parse_records() {
    let rt = Runtime::new().unwrap();
    glob!("inputs/raw-records/record*.json", |path| {
        let input = fs::read_to_string(path).unwrap();
        let key_value: KeyValue = serde_json::from_str(&input).unwrap();
        let owned_message = OwnedMessage::new(
            key_value.value,
            key_value.key,
            "my-topic".to_string(),
            rdkafka::Timestamp::CreateTime(0),
            0,
            0,
            None,
        );
        rt.block_on(async {
            assert_debug_snapshot!(KafkaRecord::parse(owned_message, &mut None).await);
        });
    });
}

#[derive(Clone, Debug, Deserialize, Hash, PartialEq, Eq, Default)]
#[serde(rename_all = "lowercase")]
struct KeyValue {
    pub key: Option<Vec<u8>>,
    pub value: Option<Vec<u8>>,
}
