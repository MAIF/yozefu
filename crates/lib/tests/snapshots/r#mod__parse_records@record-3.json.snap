---
source: crates/lib/tests/mod.rs
expression: "KafkaRecord::parse(owned_message, &mut None).await"
input_file: crates/lib/tests/inputs/raw-records/record-3.json
---
KafkaRecord {
    topic: "my-topic",
    timestamp: Some(
        0,
    ),
    partition: 0,
    offset: 0,
    headers: {},
    key_schema: Some(
        Schema {
            id: SchemaId(
                1,
            ),
            schema_type: None,
        },
    ),
    value_schema: Some(
        Schema {
            id: SchemaId(
                2,
            ),
            schema_type: None,
        },
    ),
    size: 14,
    key: Json(
        Object {},
    ),
    key_as_string: "{}",
    value: Json(
        Object {},
    ),
    value_as_string: "{}",
}
